### ISSUE TITLE
Support Frontmatter

### ISSUE THREAD

**Issue by muuvmuuv** [mdx-js/eslint-mdx#147]
### Subject of the feature

I'm currently creating a new eslint setup in my project and everything seems fine but when I lint `.mdx` files I'm getting the below warning. I couldn't find anything in the docs or if it is even already supported but frontmatter support is necessary.

```
.../src/pages/about.mdx
  1:3  error  Parsing error: Invalid left-hand side in prefix operation

> 1 | ---
    |   ^
  2 | title: 'About Me'
  3 | ---
  4 |
```

### Problem

No frontmatter support

### Expected behavior

Should ignore the frontmatter section or lint it if it is possible (indention etc.)

### Alternatives

What are the alternative solutions?  Please describe what else you have considered?

### Additional information

My eslint config:

```json
{
  "parser": "babel-eslint",
  "plugins": [
    "import"
  ],
  "extends": [
    "eslint:recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "plugin:react/recommended",
    "plugin:mdx/recommended",
    "plugin:jsx-a11y/strict",
    "plugin:prettier/recommended"
  ],
  "env": {
    "es6": true,
    "browser": true,
    "commonjs": true,
    "serviceworker": true
  },
  "parserOptions": {
    "sourceType": "module",
    "ecmaVersion": 2020
  },
  "settings": {
    "react": {
      "version": "detect"
    },
    "import/resolver": {
      "node": {
        "extensions": [
          ".js",
          ".jsx"
        ]
      }
    }
  },
  "rules": {
    "react/display-name": "off",
    "react/prop-types": "off",
    "import/order": [
      "error",
      {
        "groups": [
          [
            "builtin",
            "external",
            "internal"
          ]
        ]
      }
    ],
    "default-case": "off"
  }

```

**Comment by JounQin** [comment]
Try to use `overrides` of ESLint first.

And, a minimal runnable reproduction is required.

**Comment by kate-hall** [comment]
I'm also having this issue. I think there are flaws in the code suggested by JounQin above if anyone is banging their heads on this: `plugin/mdx:recommended` should be `plugin:mdx/recommended`

Note that neither suggested solution allowed frontmatter in MDX files to pass through the linter successfully.

**Comment by JounQin** [comment]
@kate-hall Hah, thanks for pointing out my typo.

Would you like to post a new issue with a minimal and runnable reproduction?

**Comment by madcatsu** [comment]
Adding to this issue.

I'm experiencing the same with eslint >= 7.8.x and following Eslint config:

```javascript
const OFF = 0;
const WARNING = 1;
const ERROR = 2;

module.exports = {
  root: true,
  env: {
    browser: true,
    commonjs: true,
    jest: true,
    node: true,
  },
  parser: 'babel-eslint',
  parserOptions: {
    allowImportExportEverywhere: true,
  },
  extends: ['airbnb', 'prettier', 'prettier/react', 'plugin:mdx/recommended'],
  plugins: ['react-hooks'],
  rules: {
    // Ignore certain webpack alias because it can't be resolved
    'import/no-unresolved': [
      ERROR,
      { ignore: ['^@theme', '^@docusaurus', '^@generated'] },
    ],
    'import/extensions': OFF,
    'react/jsx-closing-bracket-location': OFF, // Conflicts with Prettier.
    'react/jsx-filename-extension': OFF,
    'react-hooks/rules-of-hooks': ERROR,
    'react/prop-types': OFF
  }
};
```

I have the same problem as @muuvmuuv but apparently this actual repo doesn't have the same problem if I clone it and add some Markdown and MDX files with standard frontmatter ```---```

**Comment by JounQin** [comment]
@madcatsu Please post a new issue with runnable reproduction, thx.

**Comment by JounQin** [comment]
Please enable remark-frontmatter manually

https://github.com/mdx-js/mdx/pull/1204/files

**Comment by vincerubinetti** [comment]
@JounQin Any idea how to do this in the new ESLint flat config format? I'm not able to figure out what to do. I'm just trying to have eslint be able to lint MDX files and not be tripped up by frontmatters.

I have this in `eslint.config.mjs`:

```
const config = [
  ...compat.extends(
    "next/core-web-vitals",
    "next/typescript",
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/stylistic",
    "plugin:react-hooks/recommended",
    "plugin:jsx-a11y/recommended",
  ),
  {
    ...mdx.flat,
    plugins: { remarkFrontmatter },
    processor: mdx.createRemarkProcessor({}),
  },
]
```

as instructed by the readme, but I don't see how to specify remark plugins, i.e. to include `remark-frontmatter`.

Do you have any ideas?

**Comment by switz** [comment]
@vincerubinetti did you ever figure this out?
